Provider/src/main/java/com/spotify/confidence/ConfidenceFeatureProvider.kt

```diff
@@ -49,52 +49,59 @@ class ConfidenceFeatureProvider private constructor(
         }
     }
 
+    private fun startListeningForContext() {
+        coroutineScope.launch {
+            confidence.contextChanges
+                .collect {
+                    resolve(InitialisationStrategy.FetchAndActivate)
+                }
+        }
+    }
+
     override fun initialize(initialContext: EvaluationContext?) {
         initialContext?.let {
-            internalInitialize(
-                initialContext,
-                initialisationStrategy
-            )
+            // refresh cache with the last stored data
+            storage.read().let(providerCache::refresh)
+            if (initialisationStrategy == InitialisationStrategy.ActivateAndFetchAsync) {
+                eventHandler.publish(OpenFeatureEvents.ProviderReady)
+            }
+
+            coroutineScope.launch(networkExceptionHandler) {
+                val context = initialContext.toConfidenceContext()
+                for (entry in context.map) {
+                    confidence.putContext(entry.key, entry.value)
+                }
+                resolve(initialisationStrategy)
+                startListeningForContext()
+            }
         }
     }
 
-    private fun internalInitialize(
-        initialContext: EvaluationContext,
-        strategy: InitialisationStrategy
-    ) {
-        // refresh cache with the last stored data
-        storage.read().let(providerCache::refresh)
-        if (strategy == InitialisationStrategy.ActivateAndFetchAsync) {
-            eventHandler.publish(OpenFeatureEvents.ProviderReady)
-        }
+    private suspend fun resolve(strategy: InitialisationStrategy) {
+        try {
+            val resolveResponse = confidence.resolve(listOf())
```

Comment:
I assume we don't have `RECONCILING` support from the SDK yet?
Otherwise we would emit that before resolve, right?